### PR TITLE
Validation messages text in class public static properties (#711)

### PR DESCRIPTION
**Pull request by Erindi**

### CODE REVIEW

File: src/Validator/Rules/ValuesOfCorrectType.php [hunk 1]
```diff
@@ -253,24 +271,24 @@ static function (EnumValueDefinition $value) : string {
                 )
             );
 
-            return $suggestions ? 'Did you mean the enum value ' . Utils::orList($suggestions) . '?' : null;
+            return $suggestions ? static::$enumTypeSuggestionMessage . Utils::orList($suggestions) . '?' : null;
```
Comment (spawnia) [inline src/Validator/Rules/ValuesOfCorrectType.php:274]: I think this one should be replaced with `sprintf`, grammar may not be the same in all languages.

Comment (Erindi) [reply src/Validator/Rules/ValuesOfCorrectType.php]: Replaced.

**Comment by shmax** [comment]
Is this something to do with localization of the error messages? If so, I'm not sure I can really recommend this approach. You're adding a huge number of new public properties, expanding the maintenance surface, and spreading the feature all across the code base. I would suggest an approach that keeps the details of each class insular, centralizes the error strings for easier reference, and exposes a single point where localization can happen.

I'm sure there are a million ways to do this, but I was involved in this exact same work on another repo I used to contribute to years ago, and here's what I came up with then:

https://github.com/justinrainbow/json-schema/pull/364/files

In short:
* used an enum class to define the error codes
* all of the default error message strings are in one place
* when logging an error, instead of adding the error message on the spot, we add the error code and the vars (to later pass to sprintf)

Just some ideas to get you thinking.

**Comment by Erindi** [comment]
> Is this something to do with localization of the error messages? If so, I'm not sure I can really recommend this approach. You're adding a huge number of new public properties, expanding the maintenance surface, and spreading the feature all across the code base. I would suggest an approach that keeps the details of each class insular, centralizes the error strings for easier reference, and exposes a single point where localization can happen.
> 
> I'm sure there are a million ways to do this, but I was involved in this exact same work on another repo I used to contribute to years ago, and here's what I came up with then:
> 
> https://github.com/justinrainbow/json-schema/pull/364/files
> 
> In short:
> 
>     * used an enum class to define the error codes
> 
>     * all of the default error message strings are in one place
> 
>     * when logging an error, instead of adding the error message on the spot, we add the error code and the vars (to later pass to sprintf)
> 
> 
> Just some ideas to get you thinking.

Hi, I've created an [issue](https://github.com/webonyx/graphql-php/issues/711). Got an answer that translation for this lib is not an important case and @vladar [don't want to complicate things](https://github.com/webonyx/graphql-php/issues/711#issuecomment-668505205) because of translation and proposed a minor changes so that we had at least some opportunity to translate the text of errors. @spawnia proposed another option with also minor changes [that prevents silly errors](https://github.com/webonyx/graphql-php/issues/711#issuecomment-668653299), i agreed to this terms and did this PR to get a solution for my problem in a way acceptable to you.
I know that there are a million ways to do this better but how I thought it belonged to the category of "complicate things", for not so worthy cause. And this PR here is only because this is some fast and "better than none" solution for my project that was some kind of acceptable for you in my issue.

**Comment by spawnia** [comment]
> this is some fast and "better than none" solution

@Erindi I agree with that assessment and think we can merge this as-is.

@shmax We do not have to guarantee stability of those messages forever. If someone decides to put in the time to develop a better translation system, we can change it in the next major release.

**Comment by shmax** [comment]
@spawnia @Erindi This is not a "minor change". You touched 32 files. You added 50+ new public properties to the API surface that we will have to maintain, and eventually break once one of us takes the time to support this properly. 

But the point is that this is simply not how you localize a code base. With this approach a consumer of the library would have to figure out how to get access to each of these classes so he can set those strings, and it would do absolutely nothing for client side projects (eg. Javascript).

If @Erindi is in that much of a hurry he can just fork the repo and merge his branch there. Let's not make a mess of this one.

All that said, I'm happy to support this feature properly, but let's take the time to understand it and come up with a solution that is mature, robust, and a little more accessible to other consumers of this library. I'm free all weekend.

**Comment by shmax** [comment]
I see that we already have an `extensions` property on our `Error` class, but it doesn't currently seem to be used for anything (other than some tests that write "foo" to it).

**Comment by shmax** [comment]
Here's the 1-hour version: https://github.com/webonyx/graphql-php/pull/718

**Comment by spawnia** [comment]
I noticed that `sprintf` based internationalization is fundamentally flawed, sentence structure can differ between languages.
